### PR TITLE
64 handle images to sendreceive tofrom firebase

### PR DESCRIPTION
**Pull request by Sobell13**
1. Added the ImageBitmap Input/Output class to read/write ImageBitmap onto/from the internal storage of the phone
2. Implemented ImageBitmap profile picture as a User attribute
3. Modified the UserViewModel to handle _null_ ImageBitmap

### CODE REVIEW

**Review by marinovdeyan — CHANGES_REQUESTED** [review]
Very good changes to handle profile pictures. ImageVector was indeed not adapted for this usecase.

IMPORTANT: some modifications in build.gradle.kts seem very strange and should be checked if they are needed or not. Some changes in MainActivity have been left from the testing.

File: app/src/main/java/com/swent/suddenbump/model/user/User.kt [hunk 1]
```diff
@@ -1,12 +1,12 @@
 package com.swent.suddenbump.model.user
 
-import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.graphics.ImageBitmap
```
[the remark this answers is not in the document]
Comment (Sobell13) [reply app/src/main/java/com/swent/suddenbump/model/user/User.kt:3]: I know.

File: app/src/main/java/com/swent/suddenbump/MainActivity.kt [hunk 2]
```diff
@@ -83,11 +88,9 @@ class MainActivity : ComponentActivity() {
...
             modifier = Modifier.fillMaxSize().semantics { testTag = C.Tag.main_screen_container },
             color = MaterialTheme.colorScheme.background) {
-              SuddenBumpApp(newLocation)
-              // MapScreen(this, newLocation)
-
-              // LocationHandler(this) { location -> newLocation = location }
-              //
+              // SuddenBumpApp(newLocation)
+              val userViewModel = UserViewModel(UserRepositoryFirestore(Firebase.firestore))
+              TestComposableScreen(userViewModel)
```
[the remark this answers is not in the document]
Comment (Sobell13) [reply app/src/main/java/com/swent/suddenbump/MainActivity.kt:93]: No, it was a custom MainActivity, made only to test the new functionalities

File: build.gradle.kts [hunk 3]
```diff
@@ -3,5 +3,5 @@ plugins {
     alias(libs.plugins.androidApplication) apply false
     alias(libs.plugins.jetbrainsKotlinAndroid) apply false
     alias(libs.plugins.ktfmt) apply false
-    alias(libs.plugins.gms) apply true
+    alias(libs.plugins.gms) apply false
```
[the remark this answers is not in the document]
Comment (Sobell13) [reply build.gradle.kts:6]: It is to correct a Gradle warning at build

[the remark this answers is not in the document]
Comment (marinovdeyan) [reply build.gradle.kts:6]: okay perfect, I see which warning you are talking about. thanks!

File: app/src/main/java/com/swent/suddenbump/MainActivity.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (marinovdeyan) [reply app/src/main/java/com/swent/suddenbump/MainActivity.kt:93]: okay so please remove it when merging.

**Review by marinovdeyan — APPROVED** [review]
PR approved having my questions answered. Thanks and happy merging.

**Pull request merged**